radix-engine/src/kernel/kernel.rs

```diff
@@ -82,67 +69,35 @@ where
             current_frame: CallFrame::new_root(),
             prev_frame_stack: vec![],
             module,
-        };
+        }
+    }
 
-        // Initial authzone
-        // TODO: Move into module initialization
-        kernel
-            .execute_in_mode::<_, _, RuntimeError>(ExecutionMode::AuthModule, |api| {
-                let auth_zone = AuthZoneStackSubstate::new(
-                    vec![],
-                    auth_zone_params.virtualizable_proofs_resource_addresses,
-                    auth_zone_params.initial_proofs.into_iter().collect(),
-                );
-                let node_id = api.allocate_node_id(RENodeType::AuthZoneStack)?;
-                api.create_node(
-                    node_id,
-                    RENodeInit::AuthZoneStack(auth_zone),
-                    BTreeMap::new(),
-                )?;
-                Ok(())
-            })
-            .expect("AuthModule failed to initialize");
-
-        kernel
-            .execute_in_mode::<_, _, RuntimeError>(ExecutionMode::LoggerModule, |api| {
-                LoggerModule::initialize(api)
-            })
-            .expect("Logger failed to initialize");
-
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Resource(RADIX_TOKEN)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Resource(SYSTEM_TOKEN)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Resource(ECDSA_SECP256K1_TOKEN)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Resource(EDDSA_ED25519_TOKEN)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Resource(PACKAGE_TOKEN)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Component(EPOCH_MANAGER)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Component(CLOCK)),
-            RENodeVisibilityOrigin::Normal,
-        );
-        kernel.current_frame.add_stored_ref(
-            RENodeId::Global(GlobalAddress::Package(FAUCET_PACKAGE)),
-            RENodeVisibilityOrigin::Normal,
-        );
-
-        kernel
+    pub fn initialize(&mut self) -> Result<(), RuntimeError> {
+        self.execute_in_mode::<_, _, RuntimeError>(ExecutionMode::KernelModule, |api| {
+            KernelModuleMixer::on_init(api)
+        })
+    }
+
+    pub fn teardown(mut self) -> (KernelModuleMixer, Option<RuntimeError>) {
```

Comment:
Can we add a TODO to cleanup this interface? Feels a bit weird and my guess is that there's broken abstraction somewhere.

Reply:
Sure